snuba/pipeline/simple_pipeline.py

```diff
@@ -1,58 +1,85 @@
-)
-from snuba.pipeline.processors import (
-    _execute_entity_processors,
-    _execute_query_plan_processors,
-)
 from snuba.datasets.plans.query_plan import (
     ClickhouseQueryPlan,
     ClickhouseQueryPlanBuilder,
     QueryRunner,
 )
+from snuba.pipeline.processors import (
+    execute_entity_processors,
+    execute_pre_strategy_processors,
+)
+from snuba.pipeline.query_pipeline import (
+    QueryPlanner,
+    QueryExecutionPipeline,
+    QueryPipelineBuilder,
+)
+from snuba.query.logical import Query as LogicalQuery
 from snuba.request import Request
+from snuba.request.request_settings import RequestSettings
 from snuba.web import QueryResult
 
 
-class SimplePipeline(QueryPipeline):
+class EntityQueryPlanner(QueryPlanner[ClickhouseQueryPlan]):
     """
-    A query pipeline for a single query plan.
+    Executes the processing phase of a single plan query. Which means a
+    query based on a single entity, that would produce a query plan based
+    on a single storage.
 
-    TODO: Currently only query plan building and processing is done by the query
-    pipeline, eventually the rest of the query processing sequence will move into
-    the pipeline as well.
+    This should not be used if the plan execution strategy is then used
+    to execute the query as it executes all query processors.
+    The main use case is for subqueries.
     """
 
     def __init__(
         self,
-        request: Request,
-        runner: QueryRunner,
+        query: LogicalQuery,
+        settings: RequestSettings,
         query_plan_builder: ClickhouseQueryPlanBuilder,
+    ) -> None:
+        self.__query = query
+        self.__settings = settings
+        self.__query_plan_builder = query_plan_builder
+
+    def execute(self) -> ClickhouseQueryPlan:
+        execute_entity_processors(self.__query, self.__settings)
+
+        return self.__query_plan_builder.build_plan(self.__query, self.__settings)
+
+
+class SimpleExecutionPipeline(QueryExecutionPipeline):
+    """
+    Executes a simple (single entity) query.
+    """
+
+    def __init__(
+        self, request: Request, runner: QueryRunner, query_planner: EntityQueryPlanner,
     ):
         self.__request = request
         self.__runner = runner
-        self.__query_plan_builder = query_plan_builder
+        self.__query_planner = query_planner
 
     def execute(self) -> QueryResult:
-        # Execute entity processors
-        _execute_entity_processors(self.__request)
-
-        # Build and execute query plan
-        self.__query_plan = self.__query_plan_builder.build_plan(self.__request)
-        _execute_query_plan_processors(self.__query_plan, self.__request)
-        return self.__query_plan.execution_strategy.execute(
-            self.__query_plan.query, self.__request.settings, self.__runner
-        )
+        settings = self.__request.settings
+
+        query_plan = self.__query_planner.execute()
+        execute_pre_strategy_processors(query_plan, settings)
```

Comment:
Just want to confirm I understood this. This line won't be doing anything, right? Since the pre strategy (aka plan query) processors for a single entity query are always empty as per https://github.com/getsentry/snuba/blob/master/snuba/datasets/plans/single_storage.py#L138.
Pre strategy is a step that applies to composite queries only?

Reply:
the fact they are always empty is a coincidence. When we made the split we had use cases (we did not implement yet) for query processors that were supposed to run only once per plan. Maybe we can revisit that use case but as of now, we could have them.

Reply:
Oh now I remember. If we wanted to apply parameters that the splitter has to honor (like sampling), that has to happen per plan and not per db query.